public/src/components/epicTests/targetRegionsSelector.tsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+
+import {
+  Checkbox,
+  FormControlLabel,
+  FormGroup,
+  Theme,
+  Typography,
+  WithStyles,
+  createStyles,
+  withStyles
+} from "@material-ui/core";
+import { Region } from '../../utils/models';
+
+const styles = ({ spacing, typography }: Theme) => createStyles({
+  selectLabel: {
+    fontSize: typography.pxToRem(18),
+    fontWeight: typography.fontWeightMedium,
+    color: 'black',
+  },
+  indentedCheckbox: {
+    marginLeft: spacing(3),
+  }
+});
+
+const regionLabels = {
+  AUDCountries: 'Australia',
+  Canada: 'Canada',
+  EURCountries: 'Europe',
+  NZDCountries: 'New Zealand',
+  GBPCountries: 'the UK',
+  UnitedStates: 'the US',
+  International: 'Rest-of-world'
+}
+
+interface TargetRegionsSelectorProps extends WithStyles<typeof styles> {
+  regions: Region[],
+  onRegionsUpdate: (selectedRegions: Region[]) => void,
+  isEditable: boolean,
+}
+
+interface TargetRegionsSelectorState {
+  selectedRegions: Region[],
+  allRegions: boolean,
```

Comment:
I think we can implement this with a single field in the state, so just:
```
interface TargetRegionsSelectorState {
  selectedRegions: Region[]
}
```
This is because the other 2 fields are derived from `selectedRegions`, and it's generally preferable to avoid maintaining state based on some other state.